Make call to compilation.CompleteTrees really unconditional.

VSadov opened the pull request:
Make call to compilation.CompleteTrees really unconditional.

Fixes https://github.com/dotnet/roslyn/issues/11368

mavasani commented:
👍 Thanks!

jaredpar commented on src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs:
```diff
@@ -508,11 +508,11 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                                 }
                             }
 
+                            compilation.CompleteTrees(null);
```
In the case of an exception this method won't be called.  This meant the async queue won't follow it's desired state transition.  Will that cause deadlocks / issues down the road?  What are the guarantees we provide?  

@gafter may have input here.

jaredpar commented:
CC @dotnet/roslyn-compiler

AlekseyTs commented:
LGTM

VSadov commented:
@dotnet/roslyn-compiler  @mavasani  @gafter  Added another small fix and removed the SpinWait. 
I think, to the best of our knowledge, this should work reliably in GetDiagnostics scenario.

Added some relevant comments.

VSadov commented:
I am going to check this in since the bug randomly breaks builds. 
The PR should be considered still open for comments.

cston commented:
LGTM
